src/Business.php

```diff
@@ -26,13 +26,13 @@
      * Creates a new business.
      *
      * @param DayInterface[]     $days
-     * @param \DateTime[]        $holidays
+     * @param Holidays|null      $holidays
      * @param \DateTimeZone|null $timezone
      */
-    public function __construct(array $days, array $holidays = [], \DateTimeZone $timezone = null)
+    public function __construct(array $days, Holidays $holidays = null, \DateTimeZone $timezone = null)
```

Comment:
Indeed. How should I proceed with the type check?

If an array is passed it is wrapped in a Holidays object. If null is passed, an empty object is created. What about other types? Check in the Business constructor and throw an invalid argument exception?